src/core/manager.c

```diff
@@ -4854,9 +4854,7 @@ char* manager_taint_string(const Manager *m) {
         if (os_release_support_ended(NULL, /* quiet= */ true, NULL) > 0)
                 stage[n++] = "support-ended";
 
-        _cleanup_free_ char *destination = NULL;
-        if (readlink_malloc("/var/run", &destination) < 0 ||
-            !PATH_IN_SET(destination, "../run", "/run"))
+        if (inode_same("/run", "/var/run", AT_NO_AUTOMOUNT) <= 0)
```

Comment:
i think for the taint thing there's value in being particularly strict, i.e. only allow two rlevant strings and nothing else. 

i.e. i think a verifier (such as the tain string stuff) should be stricter than an executor (such as the $PATH generation logic). i.e. given that the verifier doesn't immediately cause stuff to fail or change, but just report weirdness, I think there's value in making it super strict i.e. check the exact strings of the symlinks